### PR TITLE
docs: Document requirements of `cdk-base`

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

List instance tag and IAM requirements for the `cdk-base` AMIgo role. Also provide CFN YAML snippets for ease of adoption.

## How to test

Does it [read](https://github.com/guardian/amigo/blob/aa-cdk-base-readme/roles/cdk-base/README.md) well?

## How can we measure success?

Adoption of this role is that much easier.

## Have we considered potential risks?

I've not checked the permissions are exhaustive...

### CODE REVIEW

File: roles/cdk-base/README.md [hunk 1]
```diff
@@ -1,34 +1,111 @@
...
-To ship logs, ensure your instance has the following tag:
+### Kinesis
+EC2 instances must have a `LogKinesisStreamName` tag.
 
-    LogKinesisStreamName
+We ship logs into the Central ELK stack via a Kinesis stream. `devx-logs` discovers this stream via the `LogKinesisStreamName` tag on the EC2 instance.
 
-set to the name of your logging Kinesis stream.
+<details>
+<summary>CFN YAML snippet</summary>
```
Comment (akash1810) [inline roles/cdk-base/README.md:20]: This renders the YAML snippet in a collapsible section on the basis that we want GuCDK to be used.

File: roles/cdk-base/README.md [hunk 2]
```diff
@@ -1,34 +1,111 @@
 # CDK Base
 
-**Note: this role is experimental. It is safe to use but the precise behaviour
-and required tags are still subject to change.**
+**Note: this role is experimental. It is safe to use but the precise behaviour and required tags are still subject to change.**
```
Comment (nicl) [inline roles/cdk-base/README.md:3]: Perhaps we should agree a line length and format to that?

File: roles/cdk-base/README.md [hunk 3]
```diff
@@ -1,34 +1,111 @@
...
+</details>
 
-    kinesis:DescribeStream
-    kinesis:PutRecord
+### IAM permissions
+EC2 instances must have the following IAM permissions:
+- `kinesis:DescribeStream` scoped to the Kinesis stream above
+- `kinesis:PutRecord` scoped to the Kinesis stream above
+- `autoscaling:DescribeAutoScalingInstances`
+- `autoscaling:DescribeAutoScalingGroups`
```
Comment (nicl) [inline roles/cdk-base/README.md:43]: These lines are not requires anymore, as instance-tag-discovery does not look up ASG tags anymore - see https://github.com/guardian/instance-tag-discovery/pull/2

File: roles/cdk-base/README.md [hunk 4]
```diff
@@ -1,34 +1,111 @@
...
+        - Effect: Allow
+          Action:
+          - kinesis:DescribeStream
+          - kinesis:PutRecord
+          Resource: !Sub 'arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:stream/${LoggingStreamName}'
+        - Effect: Allow
+          Resource: '*'
+          Action:
+          - autoscaling:DescribeAutoScalingInstances
+          - autoscaling:DescribeAutoScalingGroups
```
Comment (nicl) [inline roles/cdk-base/README.md:85]: (Ditto above.)

**Review by nicl — APPROVED** [review]
Nice! Much clearer :)

**Review by nicl — APPROVED** [review]
👍 

Only question is around `autoscaling:DescribeAutoScalingInstances` - as I'm not sure this is needed.

**Pull request merged**